test-assets/ffbc54/shortcut.js

```diff
@@ -0,0 +1,34 @@
+const defaultParams = {
+  target: 'target',
+  focusOnly: false, 
+  shortcutKey: '+', 
+  ctrlKey: false,
+  disabled: false
+};
+
+let allSettings = new Array();
+
+function shortcut(...params) {
+  allSettings = [...allSettings, ...params.map(p => { return { ...defaultParams, ...p } })];
+  document.body.addEventListener("keydown", function(event) {
+    for (const settings of allSettings || []) {
+      if (!settings.disabled) {
+        const target = document.getElementById(settings.target);
+
+        if (
+          event.key === settings.shortcutKey &&
+          (!settings.ctrlKey || event.ctrlKey) &&
+          (!settings.focusOnly || document.activeElement === target)
+        ) {
+          document.getElementById("list").innerHTML += "<li>" + target.value + "</li>";
+          target.value = "";
+          event.preventDefault();
+        }
+      }
+    }
+  });
+}
+
+function changeSetting(reference, setting, value) {
+  allSettings.find(set => set.reference === reference)[setting] = value;
+}
```

Comment:
`reference` is not a very good name in the sense that it doesn't trigger the correct bells… What about `id` (implies "short name used to uniquely identify something")?

Also, `set` is not a very good name because it is used as an abbreviation for "set"ting but can also just mean a set, and that is confusing. That reinforce the idea of renaming `allSettings` to something about shortcuts. That array is a list of shortcut definitions. Each definition of a shortcut contains several settings. Then it make sense to have
```javascript
shortcutDefinitions.find(shortcut => shortcut.id === id)[setting] = value
```
that reads much better (I think) as "look into the list of shortcut definitions, find the shortcut whose id match the one I want and update a setting on that shortcut".

---
Given the 2 use cases of that function (disabling, and adding a ctrl modifier), I think it is worth making the two corresponding functions:
```javascript
function toggleDisabled(id, value) {
  changeSetting(id, "disabled", value)
}

function toggleModifier(id, value) {
  changeSetting(id, "ctrlKey", value)
}
```
the calls in the examples will be much more explicit in what they do and won't require reader to so look the code to understand the effect.